### ISSUE TITLE
Improve Intl.CollatorOptions type definitions

### ISSUE THREAD

**Issue by ivan-palatov** [microsoft/TypeScript#53507]
# lib Update Request

## Configuration Check

My compilation *target* is `ESNEXT` and my *lib* is `ESNEXT`.

## Missing / Incorrect Definition

Interface **CollatorOptions** in **Intl** module is not strict enough. It allows for any string to be passed to `localeMatcher`, `usage`, `sensitivity` and `caseFirst` properties, which should be restricted to their respective unions of possible values.

## Sample Code

This should complain, since every value in the options isn't usable per spec, but doesnt.

```typescript
const collator = new Intl.Collator(undefined, { 
  localeMatcher: 'custom',
  usage: 'my-use-case',
  sensitivity: 'very-sensitive',
  caseFirst: 'nope' 
});
```

## Documentation Link

[MDN Intl.Collator docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator/Collator)
